connector_alndata/models/crm_lead.py

```diff
@@ -0,0 +1,303 @@
+        return content
+
+    @api.model
+    def sync_market_data(self, cr):
+        """Synchronize Market Data.
+
+        This method is used to sync market data.
+        """
+        industry_obj = self.env['res.partner.industry']
+        cr = self._cr
+        market_data = self.aln_auth_login('Markets')
+        market_ids = []
+        if market_data:
+            market_data = eval(market_data).get("value", [])
+            for market in market_data:
+                cr.execute("""
+                    Select
+                        id , name
+                    from
+                        res_partner_industry
+                    where
+                         name= '%s' or full_name='%s' and
+                         industry_type='market'
+                """ % (market.get('MarketCode'),
+                       market.get('MarketDescription')))
+                available_market = cr.fetchall()
+                if not available_market:
+                    market = industry_obj.create(
+                        {
+                            'name': market.get('MarketCode'),
+                            'full_name': market.get('MarketDescription'),
+                            'industry_type': 'market',
+                        })
+                    market_ids.append(market.id)
+        _logger.info('Created Market Ids : %s', market_ids)
+
+    @api.model
+    def sync_submarket_data(self, cr):
+        """Synchronize Submarket Data.
+
+        This method is used to sync submarket data.
+        """
+        industry_obj = self.env['res.partner.industry']
+        cr = self._cr
+        submarket_ids = []
+        submarket_data = self.aln_auth_login('Submarkets')
+        if submarket_data:
+            submarket_data = eval(submarket_data).get("value", [])
+            for submarket in submarket_data:
+                cr.execute("""
+                    Select
+                        id
+                    from
+                        res_partner_industry
+                    where name= '%s' and industry_type='sub-market'
+                """ % (submarket.get('SubMarketDescription'),))
+                available_submarket = cr.fetchone()
+                cr.execute("""
+                    Select
+                        id
+                    from
+                        res_partner_industry
+                    where name= '%s' and industry_type='market'
+                    limit 1
+                """ % (submarket.get('Market'),))
+                market_id = cr.fetchone()
+                market_id = market_id and market_id[0]
+                if not available_submarket:
+                    submarket = industry_obj.create(
+                        {
+                            'name': submarket.get('SubMarketDescription'),
+                            'parent_id': market_id,
+                            'industry_type': 'sub-market',
+                            'ref': submarket.get('SubmarketId'),
```

Comment:
Please add a method `_prepare_industry_values` on `res.partner.industry` to return this dictionary.